src/UI/Component/Item/AppointmentItem.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+/* Copyright (c) 2016 Timon Amstutz <timon.amstutz@ilub.unibe.ch> Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\UI\Component\Item;
+
+/**
+ * Interface AppointmentItem
+ * @package ILIAS\UI\Component\Panel\Listing
+ */
+interface AppointmentItem extends Item {
+	/**
+	 * Get the starting point of the appointment.
+	 * @return \ilDateTime
+	 */
+	public function getFrom();
+
+	/**
+	 * Get the ending point of the appointment.
+	 * @return \ilDateTime
+	 */
+	public function getEnd();
+
+	/**
+	 * Get the color of the calendar containing the item as color code (hex).
+	 * @return string
```

Comment:
Sorry the availability of the old interface proposal makes it harder to read. On the other hand it's easier to compare imo. As written in my initial comment I would neither work with strings or color data objects that hold hex values for colors. I would prefer to have predefined color palette (in the CSS) that can be referenced by integer IDs (1-16, 1-32, whatever number would be feasible).